modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,20 +370,107 @@ public void testGetSuccess(String id, String attributes,
     }
 
     @Test(dataProvider = "dataForGetThrowingExceptions")
-    public void testGetThrowingExceptions(String id, String attributes,
-                     String excludeAttributes) throws CharonException {
+    public void testGetThrowingExceptions(String id, String attributes, String excludeAttributes)
+            throws CharonException, BadRequestException, NotFoundException {
 
         SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
         Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
                 (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
 
         abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
-                .thenReturn(SCIM2_ME_ENDPOINT);
-        abstractResourceManager.when(() -> userManager.getUser(id, requiredAttributes)).thenReturn(null);
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        Mockito.when(userManager.getUser(id, requiredAttributes)).thenReturn(null);
 
         SCIMResponse outputScimResponse = userResourceManager.get(id, userManager, attributes, excludeAttributes);
 
         Assert.assertNull(outputScimResponse);
+
+    }
+
+    @DataProvider(name = "dataForGetUserNotFoundException")
+    public Object[][] dataToGetUserNotFoundException() {
+
+        return new Object[][]{
+                {"1234", "userName", null},
+                {"1234", "userName", "emails"}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetUserNotFoundException")
+    public void testGetUserNotFoundException(String id, String attributes, String excludeAttributes)
```

Comment:
what's the difference between the method here and the method in https://github.com/wso2/charon/pull/335/files#diff-32914d3be281f9ccb36fe8cc05d50a694f7364dbf5b6ba7623f7d29103998f38R373 ?

Reply:
"https://github.com/wso2/charon/pull/335/files#diff-32914d3be281f9ccb36fe8cc05d50a694f7364dbf5b6ba7623f7d29103998f38R373"  This is previous author,  404 has been validated using null assertion in that case, can delete the test case if we want .

Reply:
if the new test case added by you covers the previous test case, no need to have the previous test case.